src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -81,16 +84,16 @@ class PosHashTable {
   // of the offset does not limit the number of rows in the partition but the number of distinct values. If we end up
   // with a partition that has more values, the partitioning algorithm is at fault.
   using Offset = uint32_t;
-  using HashTable = ska::bytell_hash_map<HashedType, Offset>;
+  using HashTable = tsl::robin_pg_map<HashedType, Offset, std::hash<HashedType>, std::equal_to<HashedType>, PolymorphicAllocator<std::pair<HashedType, Offset>>>;
```

Comment:
(1) new hash map

Reply:
https://github.com/hyrise/hyrise/compare/61bd7a4..b5a9759

has no big impact

```
+----------++----------+---------+--------++----------+----------+--------+---------+
| Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
|          ||      old |     new |        ||      old |      new |        |         |
+----------++----------+---------+--------++----------+----------+--------+---------+
| TPC-H 01 ||   1651.5 |  1667.5 |   +1%  ||     0.61 |     0.60 |   -1%  |  0.5821 |
| TPC-H 02 ||     11.8 |    11.2 |   -5%  ||    84.50 |    89.26 |   +6%  |  0.0000 |
| TPC-H 03 ||    178.4 |   178.6 |   +0%  ||     5.61 |     5.60 |   -0%  |  0.5016 |
| TPC-H 04 ||    152.0 |   143.5 |   -6%  ||     6.58 |     6.97 |   +6%  |  0.0000 |
| TPC-H 05 ||    470.9 |   457.3 |   -3%  ||     2.12 |     2.19 |   +3%  |  0.0000 |
| TPC-H 06 ||      7.9 |     8.0 |   +1%  ||   126.54 |   125.72 |   -1%  |  0.0116 |
| TPC-H 07 ||    126.2 |   121.1 |   -4%  ||     7.93 |     8.26 |   +4%  |  0.0001 |
| TPC-H 08 ||    156.3 |   150.7 |   -4%  ||     6.40 |     6.64 |   +4%  |  0.0000 |
| TPC-H 09 ||   1013.1 |  1016.4 |   +0%  ||     0.99 |     0.98 |   -0%  |  0.3955 |
| TPC-H 10 ||    511.9 |   491.1 |   -4%  ||     1.95 |     2.04 |   +4%  |  0.0000 |
| TPC-H 11 ||     16.5 |    16.5 |   +0%  ||    60.71 |    60.69 |   -0%  |  0.8161 |
| TPC-H 12 ||     90.0 |    85.4 |   -5%  ||    11.11 |    11.71 |   +5%  |  0.0000 |
| TPC-H 13 ||    705.9 |   721.5 |   +2%  ||     1.42 |     1.39 |   -2%  |  0.0000 |
| TPC-H 14 ||     38.8 |    31.2 |  -20%  ||    25.77 |    32.06 |  +24%  |  0.0000 |
| TPC-H 15 ||     17.6 |    17.5 |   -0%  ||    56.89 |    57.00 |   +0%  |  0.0510 |
| TPC-H 16 ||    178.3 |   174.0 |   -2%  ||     5.61 |     5.75 |   +3%  |  0.0000 |
| TPC-H 17 ||     54.4 |    54.3 |   -0%  ||    18.39 |    18.42 |   +0%  |  0.4706 |
| TPC-H 18 ||   1312.0 |  1312.7 |   +0%  ||     0.76 |     0.76 |   -0%  |  0.9296 |
| TPC-H 19 ||     75.3 |    75.3 |   +0%  ||    13.28 |    13.28 |   -0%  |  0.9126 |
| TPC-H 20 ||     31.7 |    30.2 |   -5%  ||    31.54 |    33.13 |   +5%  |  0.0000 |
| TPC-H 21 ||    822.2 |   850.2 |   +3%  ||     1.22 |     1.18 |   -3%  |  0.0000 |
| TPC-H 22 ||    126.3 |   130.7 |   +3%  ||     7.92 |     7.65 |   -3%  |  0.0000 |
+----------++----------+---------+--------++----------+----------+--------+---------+
| Sum      ||   7748.8 |  7744.7 |   -0%  ||          |          |        |         |
| Geomean  ||          |         |        ||          |          |   +2%  |         |
+----------++----------+---------+--------++----------+----------+--------+---------+
```

Reply:
Q14 was one of the queries that I was looking at. `+24%` looks like something that's worth having. @Bouncner, how do you feel about using a different hash map?

Reply:
Actually, I am in favor of it. First, our hash map appears to be less and less maintained (we already needed to remove the usage of useful C++17 function when moving away from std, this might add up over time) and tessil’s maps are very well maintained.
I remember ghat tsl wasn’t that fast back then, but if it’s only slightly faster now, I am in favor of changing it.

Reply:
The fact that `bytell_hash_map` didn't work with the allocator right away is another reason to change this. @j-tr, could you look if tsl has any performance impact in the aggregate, too?

Reply:
https://github.com/hyrise/hyrise/compare/61bd7a4..ae1b0a0

tsl hash map for aggregate_hash seems to mostly lower performance.

```
+----------++----------+---------+--------++----------+----------+--------+---------+
| Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
|          ||      old |     new |        ||      old |      new |        |         |
+----------++----------+---------+--------++----------+----------+--------+---------+
| TPC-H 01 ||   1651.5 |  1733.9 |   +5%  ||     0.61 |     0.58 |   -5%  |  0.0073 |
| TPC-H 02 ||     11.8 |    11.6 |   -2%  ||    84.50 |    86.39 |   +2%  |  0.0677 |
| TPC-H 03 ||    178.4 |   178.1 |   -0%  ||     5.61 |     5.61 |   +0%  |  0.4204 |
| TPC-H 04 ||    152.0 |   153.5 |   +1%  ||     6.58 |     6.52 |   -1%  |  0.0000 |
| TPC-H 05 ||    470.9 |   467.5 |   -1%  ||     2.12 |     2.14 |   +1%  |  0.0906 |
| TPC-H 06 ||      7.9 |     8.0 |   +2%  ||   126.54 |   124.23 |   -2%  |  0.0000 |
| TPC-H 07 ||    126.2 |   125.9 |   -0%  ||     7.93 |     7.94 |   +0%  |  0.8507 |
| TPC-H 08 ||    156.3 |   154.5 |   -1%  ||     6.40 |     6.47 |   +1%  |  0.0139 |
| TPC-H 09 ||   1013.1 |  1044.3 |   +3%  ||     0.99 |     0.96 |   -3%  |  0.0000 |
| TPC-H 10 ||    511.9 |   507.7 |   -1%  ||     1.95 |     1.97 |   +1%  |  0.4646 |
| TPC-H 11 ||     16.5 |    16.0 |   -3%  ||    60.71 |    62.34 |   +3%  |  0.0000 |
| TPC-H 12 ||     90.0 |    89.1 |   -1%  ||    11.11 |    11.22 |   +1%  |  0.0000 |
| TPC-H 13 ||    705.9 |   840.6 |  +19%  ||     1.42 |     1.19 |  -16%  |  0.0000 |
| TPC-H 14 ||     38.8 |    38.1 |   -2%  ||    25.77 |    26.26 |   +2%  |  0.0000 |
| TPC-H 15 ||     17.6 |    17.0 |   -3%  ||    56.89 |    58.91 |   +4%  |  0.0000 |
| TPC-H 16 ||    178.3 |   176.3 |   -1%  ||     5.61 |     5.67 |   +1%  |  0.0040 |
| TPC-H 17 ||     54.4 |    55.0 |   +1%  ||    18.39 |    18.19 |   -1%  |  0.0000 |
| TPC-H 18 ||   1312.0 |  1217.0 |   -7%  ||     0.76 |     0.82 |   +8%  |  0.0000 |
| TPC-H 19 ||     75.3 |    75.4 |   +0%  ||    13.28 |    13.25 |   -0%  |  0.0476 |
| TPC-H 20 ||     31.7 |    31.7 |   -0%  ||    31.54 |    31.58 |   +0%  |  0.6923 |
| TPC-H 21 ||    822.2 |   840.9 |   +2%  ||     1.22 |     1.19 |   -2%  |  0.0006 |
| TPC-H 22 ||    126.3 |   125.5 |   -1%  ||     7.92 |     7.97 |   +1%  |  0.0000 |
+----------++----------+---------+--------++----------+----------+--------+---------+
| Sum      ||   7748.8 |  7907.5 |   +2%  ||          |          |        |         |
| Geomean  ||          |         |        ||          |          |   -0%  |         |
+----------++----------+---------+--------++----------+----------+--------+---------+
```

Reply:
```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||    810.9 |   813.5 |   +0%  ||     1.23 |     1.23 |   -0%  |  0.6034 |
+| TPC-H 02 ||      5.2 |     4.9 |   -5%  ||   193.44 |   203.69 |   +5%  |  0.0000 |
 | TPC-H 03 ||     83.8 |    83.1 |   -1%  ||    11.93 |    12.03 |   +1%  |  0.0000 |
 | TPC-H 04 ||     68.4 |    68.7 |   +0%  ||    14.62 |    14.57 |   -0%  |  0.0000 |
 | TPC-H 05 ||    232.3 |   229.3 |   -1%  ||     4.30 |     4.36 |   +1%  |  0.0011 |
+| TPC-H 06 ||      3.5 |     3.2 |   -9%  ||   282.34 |   309.81 |  +10%  |  0.0000 |
 | TPC-H 07 ||     58.5 |    58.2 |   -1%  ||    17.08 |    17.19 |   +1%  |  0.5031 |
 | TPC-H 08 ||     76.5 |    76.0 |   -1%  ||    13.07 |    13.15 |   +1%  |  0.2273 |
 | TPC-H 09 ||    517.0 |   519.7 |   +1%  ||     1.93 |     1.92 |   -1%  |  0.1708 |
 | TPC-H 10 ||    262.4 |   256.2 |   -2%  ||     3.81 |     3.90 |   +2%  |  0.0325 |
+| TPC-H 11 ||      8.9 |     8.1 |  -10%  ||   112.24 |   124.08 |  +11%  |  0.0000 |
 | TPC-H 12 ||     43.6 |    43.9 |   +1%  ||    22.96 |    22.78 |   -1%  |  0.0000 |
-| TPC-H 13 ||    351.5 |   369.3 |   +5%  ||     2.85 |     2.71 |   -5%  |  0.0000 |
 | TPC-H 14 ||     21.1 |    21.6 |   +2%  ||    47.39 |    46.33 |   -2%  |  0.0000 |
 | TPC-H 15 ||      7.7 |     8.0 |   +5%  ||   130.28 |   124.66 |   -4%  |  0.0000 |
 | TPC-H 16 ||     93.8 |    93.1 |   -1%  ||    10.66 |    10.74 |   +1%  |  0.0166 |
 | TPC-H 17 ||     19.5 |    19.6 |   +1%  ||    51.38 |    51.07 |   -1%  |  0.0020 |
 | TPC-H 18 ||    697.6 |   669.6 |   -4%  ||     1.43 |     1.49 |   +4%  |  0.0012 |
 | TPC-H 19 ||     31.4 |    31.1 |   -1%  ||    31.89 |    32.13 |   +1%  |  0.0000 |
 | TPC-H 20 ||     16.5 |    16.5 |   -0%  ||    60.63 |    60.68 |   +0%  |  0.7204 |
 | TPC-H 21 ||    420.9 |   418.5 |   -1%  ||     2.38 |     2.39 |   +1%  |  0.2043 |
 | TPC-H 22 ||     50.5 |    50.9 |   +1%  ||    19.79 |    19.66 |   -1%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||   3881.3 |  3862.9 |   -0%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   +1%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```

Doesn't look as bad on nemea. Given that tsl seems to be better maintained and has features like heterogenous lookup (which I've been missing a while ago), I'd say we change the two uses of bytell_hash_map to this one.

Reply:
@Bouncner - I ran more benchmarks and it does not look like robin_map is consistently better. For TPC-H SF10, it slightly increases the overall runtime. Furthermore, in the hash-based aggregate, it is significantly fast than the current map in one location and slower in another. Thus, I will not open a PR for this part.


===========

**System**
<details>
<summary>nemea - click to expand</summary>

| property | value |
| -- | -- |
| Hostname | nemea |
| CPU | Intel(R) Xeon(R) Platinum 8180 CPU @ 2.50GHz |
| Memory | 1.1TB |
| numactl | nodebind: 0  |
| numactl | membind: 0  |
</details>

**Commit Info and Build Time**
| commit | date | message | build time |
| -- | -- | -- | -- |
| dcd55668b | 16.10.2020 23:04 | JoinHash: Materialize smaller side first (#2239) | real 265.90 user 2438.68 sys 148.35|
| 3004a139f | 19.10.2020 10:40 | Remove PMR 2 | real 266.95 user 2440.98 sys 148.25|


**hyriseBenchmarkTPCH - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: +0%
 ||
Geometric mean of throughput changes: +2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_dcd55668b1db4e94834f448cc5c04498c50dda00_st.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_3004a139fc52f120d69984ae7b96b89a191e71f1_st.json |
 +--------------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                         | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                         |
 |  benchmark_mode          | Ordered                                                                                                                                | Ordered                                                                                                                                |
 |  build_type              | release                                                                                                                                | release                                                                                                                                |
 |  chunk_size              | 65535                                                                                                                                  | 65535                                                                                                                                  |
 |  clients                 | 1                                                                                                                                      | 1                                                                                                                                      |
 |  compiler                | gcc 9.2                                                                                                                                | gcc 9.2                                                                                                                                |
 |  cores                   | 0                                                                                                                                      | 0                                                                                                                                      |
 |  date                    | 2020-10-17 00:49:55                                                                                                                    | 2020-10-19 10:48:38                                                                                                                    |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}                                                                                                | {'default': {'encoding': 'Dictionary'}}                                                                                                |
 |  indexes                 | False                                                                                                                                  | False                                                                                                                                  |
 |  max_duration            | 60000000000                                                                                                                            | 60000000000                                                                                                                            |
 |  max_runs                | -1                                                                                                                                     | -1                                                                                                                                     |
 |  scale_factor            | 1.0                                                                                                                                    | 1.0                                                                                                                                    |
 |  time_unit               | ns                                                                                                                                     | ns                                                                                                                                     |
 |  use_prepared_statements | False                                                                                                                                  | False                                                                                                                                  |
 |  using_scheduler         | False                                                                                                                                  | False                                                                                                                                  |
 |  verify                  | False                                                                                                                                  | False                                                                                                                                  |
 |  warmup_duration         | 0                                                                                                                                      | 0                                                                                                                                      |
 +--------------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||    828.8 |   847.8 |   +2%  ||     1.21 |     1.18 |   -2%  |  0.0030 |
+| TPC-H 02 ||      4.8 |     4.0 |  -17%  ||   207.59 |   250.02 |  +20%  |  0.0000 |
 | TPC-H 03 ||     78.8 |    76.8 |   -3%  ||    12.68 |    13.02 |   +3%  |  0.0000 |
+| TPC-H 04 ||     68.8 |    59.7 |  -13%  ||    14.54 |    16.75 |  +15%  |  0.0000 |
+| TPC-H 05 ||    210.4 |   195.5 |   -7%  ||     4.75 |     5.11 |   +8%  |  0.0000 |
 | TPC-H 06 ||      3.4 |     3.2 |   -4%  ||   296.04 |   308.16 |   +4%  |  0.0000 |
+| TPC-H 07 ||     57.2 |    52.9 |   -7%  ||    17.49 |    18.90 |   +8%  |  0.0000 |
+| TPC-H 08 ||     62.6 |    58.9 |   -6%  ||    15.97 |    16.97 |   +6%  |  0.0000 |
 | TPC-H 09 ||    480.6 |   490.1 |   +2%  ||     2.08 |     2.04 |   -2%  |  0.0000 |
 | TPC-H 10 ||    166.8 |   165.4 |   -1%  ||     6.00 |     6.04 |   +1%  |  0.1717 |
+| TPC-H 11 ||      9.1 |     8.5 |   -6%  ||   110.01 |   117.44 |   +7%  |  0.0000 |
 | TPC-H 12 ||     44.1 |    42.8 |   -3%  ||    22.69 |    23.38 |   +3%  |  0.0000 |
-| TPC-H 13 ||    359.3 |   389.0 |   +8%  ||     2.78 |     2.57 |   -8%  |  0.0000 |
+| TPC-H 14 ||     21.2 |    18.7 |  -12%  ||    47.07 |    53.46 |  +14%  |  0.0000 |
 | TPC-H 15 ||      7.9 |     8.0 |   +2%  ||   127.26 |   124.34 |   -2%  |  0.0000 |
 | TPC-H 16 ||     75.6 |    75.9 |   +0%  ||    13.23 |    13.18 |   -0%  |  0.0000 |
-| TPC-H 17 ||     16.7 |    17.8 |   +6%  ||    59.74 |    56.29 |   -6%  |  0.0000 |
 | TPC-H 18 ||    715.5 |   689.1 |   -4%  ||     1.40 |     1.45 |   +4%  |  0.0004 |
-| TPC-H 19 ||     29.2 |    30.9 |   +6%  ||    34.22 |    32.33 |   -6%  |  0.0000 |
+| TPC-H 20 ||     16.2 |    15.4 |   -5%  ||    61.59 |    65.03 |   +6%  |  0.0000 |
-| TPC-H 21 ||    354.5 |   376.0 |   +6%  ||     2.82 |     2.66 |   -6%  |  0.0000 |
-| TPC-H 22 ||     33.3 |    35.4 |   +6%  ||    30.02 |    28.22 |   -6%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||   3644.8 |  3661.9 |   +0%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   +2%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - single-threaded, SF 0.01**
<details>
<summary>
Sum of avg. item runtimes: -0%
 ||
Geometric mean of throughput changes: +4%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_dcd55668b1db4e94834f448cc5c04498c50dda00_st_s01.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_3004a139fc52f120d69984ae7b96b89a191e71f1_st_s01.json |
 +--------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                             | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                             |
 |  benchmark_mode          | Ordered                                                                                                                                    | Ordered                                                                                                                                    |
 |  build_type              | release                                                                                                                                    | release                                                                                                                                    |
 |  chunk_size              | 65535                                                                                                                                      | 65535                                                                                                                                      |
 |  clients                 | 1                                                                                                                                          | 1                                                                                                                                          |
 |  compiler                | gcc 9.2                                                                                                                                    | gcc 9.2                                                                                                                                    |
 |  cores                   | 0                                                                                                                                          | 0                                                                                                                                          |
 |  date                    | 2020-10-17 01:12:13                                                                                                                        | 2020-10-19 11:10:56                                                                                                                        |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}                                                                                                    | {'default': {'encoding': 'Dictionary'}}                                                                                                    |
 |  indexes                 | False                                                                                                                                      | False                                                                                                                                      |
 |  max_duration            | 60000000000                                                                                                                                | 60000000000                                                                                                                                |
 |  max_runs                | -1                                                                                                                                         | -1                                                                                                                                         |
 |  scale_factor            | 0.009999999776482582                                                                                                                       | 0.009999999776482582                                                                                                                       |
 |  time_unit               | ns                                                                                                                                         | ns                                                                                                                                         |
 |  use_prepared_statements | False                                                                                                                                      | False                                                                                                                                      |
 |  using_scheduler         | False                                                                                                                                      | False                                                                                                                                      |
 |  verify                  | False                                                                                                                                      | False                                                                                                                                      |
 |  warmup_duration         | 0                                                                                                                                          | 0                                                                                                                                          |
 +--------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||      7.3 |     7.4 |   +2%  ||   136.64 |   134.48 |   -2%  |  0.0000 |
 | TPC-H 02 ||      0.5 |     0.5 |   +1%  ||  1843.23 |  1824.11 |   -1%  |  0.0181 |
+| TPC-H 03 ||      0.9 |     0.7 |  -16%  ||  1126.68 |  1344.86 |  +19%  |  0.0000 |
+| TPC-H 04 ||      0.5 |     0.4 |  -22%  ||  1928.75 |  2456.80 |  +27%  |  0.0000 |
+| TPC-H 05 ||      1.2 |     1.0 |  -20%  ||   808.20 |  1005.69 |  +24%  |  0.0000 |
 | TPC-H 06 ||      0.1 |     0.1 |   -1%  ||  9384.19 |  9494.38 |   +1%  |  0.0000 |
 | TPC-H 07 ||      1.2 |     1.2 |   +0%  ||   803.13 |   801.90 |   -0%  |  0.8888 |
 | TPC-H 08 ||     14.7 |    15.0 |   +2%  ||    67.83 |    66.55 |   -2%  |  0.0135 |
+| TPC-H 09 ||      2.4 |     2.2 |   -6%  ||   421.19 |   447.05 |   +6%  |  0.0000 |
+| TPC-H 10 ||      0.9 |     0.8 |  -13%  ||  1106.73 |  1278.69 |  +16%  |  0.0000 |
 | TPC-H 11 ||      0.3 |     0.3 |   +5%  ||  3398.24 |  3246.17 |   -4%  |  0.0000 |
 | TPC-H 12 ||      0.6 |     0.6 |   -3%  ||  1593.53 |  1634.91 |   +3%  |  0.0000 |
 | TPC-H 13 ||      2.2 |     2.3 |   +4%  ||   458.07 |   441.08 |   -4%  |  0.0000 |
+| TPC-H 14 ||      0.4 |     0.3 |  -13%  ||  2847.46 |  3258.49 |  +14%  |  0.0000 |
 | TPC-H 15 ||      1.1 |     1.1 |   +3%  ||   897.59 |   873.01 |   -3%  |  0.0000 |
 | TPC-H 16 ||      2.0 |     2.0 |   +2%  ||   495.73 |   488.00 |   -2%  |  0.0000 |
+| TPC-H 17 ||      0.3 |     0.3 |   -9%  ||  3428.06 |  3764.56 |  +10%  |  0.0000 |
 | TPC-H 18 ||      2.7 |     2.7 |   +2%  ||   373.30 |   366.36 |   -2%  |  0.0000 |
 | TPC-H 19 ||      5.5 |     5.6 |   +2%  ||   183.34 |   179.50 |   -2%  |  0.0000 |
 | TPC-H 20 ||      2.3 |     2.4 |   +1%  ||   427.15 |   423.73 |   -1%  |  0.0000 |
+| TPC-H 21 ||      1.6 |     1.5 |   -9%  ||   608.14 |   669.06 |  +10%  |  0.0000 |
 | TPC-H 22 ||      1.1 |     1.1 |   +2%  ||   895.41 |   878.61 |   -2%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||     50.0 |    49.7 |   -0%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   +4%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - single-threaded, SF 10**
<details>
<summary>
Sum of avg. item runtimes: +5%
 ||
Geometric mean of throughput changes: -2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_dcd55668b1db4e94834f448cc5c04498c50dda00_st_s10.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_3004a139fc52f120d69984ae7b96b89a191e71f1_st_s10.json |
 +--------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                             | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                             |
 |  benchmark_mode          | Ordered                                                                                                                                    | Ordered                                                                                                                                    |
 |  build_type              | release                                                                                                                                    | release                                                                                                                                    |
 |  chunk_size              | 65535                                                                                                                                      | 65535                                                                                                                                      |
 |  clients                 | 1                                                                                                                                          | 1                                                                                                                                          |
 |  compiler                | gcc 9.2                                                                                                                                    | gcc 9.2                                                                                                                                    |
 |  cores                   | 0                                                                                                                                          | 0                                                                                                                                          |
 |  date                    | 2020-10-17 01:34:16                                                                                                                        | 2020-10-19 11:33:00                                                                                                                        |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}                                                                                                    | {'default': {'encoding': 'Dictionary'}}                                                                                                    |
 |  indexes                 | False                                                                                                                                      | False                                                                                                                                      |
 |  max_duration            | 60000000000                                                                                                                                | 60000000000                                                                                                                                |
 |  max_runs                | -1                                                                                                                                         | -1                                                                                                                                         |
 |  scale_factor            | 10.0                                                                                                                                       | 10.0                                                                                                                                       |
 |  time_unit               | ns                                                                                                                                         | ns                                                                                                                                         |
 |  use_prepared_statements | False                                                                                                                                      | False                                                                                                                                      |
 |  using_scheduler         | False                                                                                                                                      | False                                                                                                                                      |
 |  verify                  | False                                                                                                                                      | False                                                                                                                                      |
 |  warmup_duration         | 0                                                                                                                                          | 0                                                                                                                                          |
 +--------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||   9085.2 |  9197.1 |   +1%  ||     0.11 |     0.11 |   -1%  |       ˅ |
+| TPC-H 02 ||     50.5 |    45.5 |  -10%  ||    19.79 |    21.97 |  +11%  |  0.0000 |
-| TPC-H 03 ||   1448.9 |  1549.5 |   +7%  ||     0.69 |     0.65 |   -6%  |  0.0000 |
-| TPC-H 04 ||   1552.1 |  1752.2 |  +13%  ||     0.64 |     0.57 |  -11%  |  0.0000 |
-| TPC-H 05 ||   3139.0 |  3421.6 |   +9%  ||     0.32 |     0.29 |   -8%  |  0.0000 |
 | TPC-H 06 ||     35.8 |    35.5 |   -1%  ||    27.91 |    28.14 |   +1%  |  0.0441 |
 | TPC-H 07 ||    831.4 |   834.9 |   +0%  ||     1.20 |     1.20 |   -0%  |  0.0294 |
 | TPC-H 08 ||    566.1 |   577.0 |   +2%  ||     1.77 |     1.73 |   -2%  |  0.0000 |
 | TPC-H 09 ||   6897.1 |  7145.6 |   +4%  ||     0.14 |     0.14 |   -3%  |       ˅ |
-| TPC-H 10 ||   2742.5 |  2894.7 |   +6%  ||     0.36 |     0.35 |   -5%  |  0.0000 |
+| TPC-H 11 ||    129.4 |   121.8 |   -6%  ||     7.73 |     8.21 |   +6%  |  0.0000 |
-| TPC-H 12 ||    685.0 |   725.8 |   +6%  ||     1.46 |     1.38 |   -6%  |  0.0000 |
-| TPC-H 13 ||   5417.0 |  5864.7 |   +8%  ||     0.18 |     0.17 |   -8%  |  0.0000 |
 | TPC-H 14 ||    233.7 |   238.2 |   +2%  ||     4.28 |     4.20 |   -2%  |  0.0000 |
 | TPC-H 15 ||     97.1 |    99.8 |   +3%  ||    10.30 |    10.02 |   -3%  |  0.0000 |
-| TPC-H 16 ||    909.1 |   964.4 |   +6%  ||     1.10 |     1.04 |   -6%  |  0.0000 |
 | TPC-H 17 ||    256.4 |   258.8 |   +1%  ||     3.90 |     3.86 |   -1%  |  0.0000 |
 | TPC-H 18 ||  11887.0 | 12328.2 |   +4%  ||     0.08 |     0.08 |   -4%  |       ˅ |
 | TPC-H 19 ||    316.5 |   319.6 |   +1%  ||     3.16 |     3.13 |   -1%  |  0.0002 |
+| TPC-H 20 ||    176.8 |   167.1 |   -6%  ||     5.65 |     5.98 |   +6%  |  0.0000 |
-| TPC-H 21 ||   5687.3 |  6026.9 |   +6%  ||     0.18 |     0.17 |   -6%  |  0.0004 |
 | TPC-H 22 ||    478.1 |   485.6 |   +2%  ||     2.09 |     2.06 |   -2%  |  0.0000 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
-| Sum      ||  52622.1 | 55054.7 |   +5%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   -2%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 |          || ˅ Insufficient number of runs for p-value calculation                 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCH - multi-threaded (50 clients)**
<details>
<summary>
Sum of avg. item runtimes: +1%
 ||
Geometric mean of throughput changes: -3%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_dcd55668b1db4e94834f448cc5c04498c50dda00_mt.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCH_3004a139fc52f120d69984ae7b96b89a191e71f1_mt.json |
 +-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                         | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                         |
 |  benchmark_mode               | Ordered                                                                                                                                | Ordered                                                                                                                                |
 |  build_type                   | release                                                                                                                                | release                                                                                                                                |
 |  chunk_size                   | 65535                                                                                                                                  | 65535                                                                                                                                  |
 |  clients                      | 50                                                                                                                                     | 50                                                                                                                                     |
 |  compiler                     | gcc 9.2                                                                                                                                | gcc 9.2                                                                                                                                |
 |  cores                        | 0                                                                                                                                      | 0                                                                                                                                      |
 |  date                         | 2020-10-17 01:59:44                                                                                                                    | 2020-10-19 11:58:26                                                                                                                    |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                | {'default': {'encoding': 'Dictionary'}}                                                                                                |
 |  indexes                      | False                                                                                                                                  | False                                                                                                                                  |
 |  max_duration                 | 60000000000                                                                                                                            | 60000000000                                                                                                                            |
 |  max_runs                     | -1                                                                                                                                     | -1                                                                                                                                     |
 |  scale_factor                 | 1.0                                                                                                                                    | 1.0                                                                                                                                    |
 |  time_unit                    | ns                                                                                                                                     | ns                                                                                                                                     |
 |  use_prepared_statements      | False                                                                                                                                  | False                                                                                                                                  |
 |  using_scheduler              | True                                                                                                                                   | True                                                                                                                                   |
 |  utilized_cores_per_numa_node | [56, 0, 0, 0]                                                                                                                          | [56, 0, 0, 0]                                                                                                                          |
 |  verify                       | False                                                                                                                                  | False                                                                                                                                  |
 |  warmup_duration              | 0                                                                                                                                      | 0                                                                                                                                      |
 +-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||   2378.5 |  2346.7 |   -1%  ||    20.55 |    20.83 |   +1%  |  0.5076 |
-| TPC-H 02 ||     16.1 |    33.2 | +106%  ||  2363.12 |  1255.90 |  -47%  |  0.0000 |
-| TPC-H 03 ||    419.6 |   444.4 |   +6%  ||   117.24 |   110.15 |   -6%  |  0.0009 |
 | TPC-H 04 ||    262.6 |   255.9 |   -3%  ||   186.30 |   190.70 |   +2%  |  0.0767 |
 | TPC-H 05 ||   1245.8 |  1282.3 |   +3%  ||    39.29 |    37.83 |   -4%  |  0.3095 |
-| TPC-H 06 ||     20.6 |    22.0 |   +7%  ||  2031.67 |  1868.43 |   -8%  |  0.0000 |
+| TPC-H 07 ||    240.6 |   224.3 |   -7%  ||   202.76 |   217.38 |   +7%  |  0.0000 |
+| TPC-H 08 ||    172.6 |   160.1 |   -7%  ||   281.27 |   302.82 |   +8%  |  0.0000 |
 | TPC-H 09 ||   2372.9 |  2419.1 |   +2%  ||    20.35 |    19.95 |   -2%  |  0.5889 |
-| TPC-H 10 ||    842.4 |   884.6 |   +5%  ||    58.61 |    55.73 |   -5%  |  0.0006 |
 | TPC-H 11 ||     66.6 |    65.8 |   -1%  ||   701.76 |   709.25 |   +1%  |  0.0159 |
 | TPC-H 12 ||    131.3 |   136.2 |   +4%  ||   366.55 |   353.54 |   -4%  |  0.0000 |
 | TPC-H 13 ||   2224.0 |  2249.3 |   +1%  ||    21.93 |    21.15 |   -4%  |  0.6803 |
-| TPC-H 14 ||    137.8 |   146.0 |   +6%  ||   349.91 |   330.80 |   -5%  |  0.0000 |
 | TPC-H 15 ||     52.1 |    51.3 |   -1%  ||   880.40 |   892.57 |   +1%  |  0.0000 |
 | TPC-H 16 ||    389.3 |   391.1 |   +0%  ||   125.75 |   125.63 |   -0%  |  0.6658 |
 | TPC-H 17 ||     55.9 |    55.3 |   -1%  ||   827.16 |   836.18 |   +1%  |  0.0036 |
 | TPC-H 18 ||   6122.1 |  6099.5 |   -0%  ||     7.70 |     7.70 |   +0%  |  0.9109 |
 | TPC-H 19 ||     92.0 |    92.4 |   +0%  ||   517.01 |   514.53 |   -0%  |  0.4948 |
+| TPC-H 20 ||     53.0 |    49.1 |   -7%  ||   866.24 |   930.20 |   +7%  |  0.0000 |
 | TPC-H 21 ||   1307.7 |  1340.7 |   +3%  ||    37.39 |    36.19 |   -3%  |  0.4938 |
 | TPC-H 22 ||    251.4 |   242.7 |   -3%  ||   193.91 |   200.78 |   +4%  |  0.0304 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||  18854.9 | 18992.1 |   +1%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   -3%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCDS - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: -2%
 ||
Geometric mean of throughput changes: +7%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview--------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter        | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCDS_dcd55668b1db4e94834f448cc5c04498c50dda00_st.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCDS_3004a139fc52f120d69984ae7b96b89a191e71f1_st.json |
 +------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH        | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                          | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                          |
 |  benchmark_mode  | Ordered                                                                                                                                 | Ordered                                                                                                                                 |
 |  build_type      | release                                                                                                                                 | release                                                                                                                                 |
 |  chunk_size      | 65535                                                                                                                                   | 65535                                                                                                                                   |
 |  clients         | 1                                                                                                                                       | 1                                                                                                                                       |
 |  compiler        | gcc 9.2                                                                                                                                 | gcc 9.2                                                                                                                                 |
 |  cores           | 0                                                                                                                                       | 0                                                                                                                                       |
 |  date            | 2020-10-17 02:22:14                                                                                                                     | 2020-10-19 12:20:56                                                                                                                     |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}                                                                                                 | {'default': {'encoding': 'Dictionary'}}                                                                                                 |
 |  indexes         | False                                                                                                                                   | False                                                                                                                                   |
 |  max_duration    | 60000000000                                                                                                                             | 60000000000                                                                                                                             |
 |  max_runs        | -1                                                                                                                                      | -1                                                                                                                                      |
 |  time_unit       | ns                                                                                                                                      | ns                                                                                                                                      |
 |  using_scheduler | False                                                                                                                                   | False                                                                                                                                   |
 |  verify          | False                                                                                                                                   | False                                                                                                                                   |
 |  warmup_duration | 0                                                                                                                                       | 0                                                                                                                                       |
 +------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |     new |        ||      old |      new |        |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
+| 01      ||     28.8 |    26.1 |   -9%  ||    34.73 |    38.33 |  +10%  |  0.0000 |
+| 03      ||      6.7 |     5.2 |  -22%  ||   150.19 |   191.36 |  +27%  |  0.0000 |
+| 06      ||     18.3 |    16.3 |  -11%  ||    54.73 |    61.46 |  +12%  |  0.0000 |
+| 07      ||     31.6 |    29.8 |   -6%  ||    31.69 |    33.59 |   +6%  |  0.0000 |
 | 09      ||    111.3 |   106.6 |   -4%  ||     8.99 |     9.38 |   +4%  |  0.0000 |
+| 10      ||     19.2 |    17.0 |  -12%  ||    52.18 |    58.98 |  +13%  |  0.0000 |
+| 13      ||    126.3 |   116.8 |   -7%  ||     7.92 |     8.56 |   +8%  |  0.0000 |
 | 15      ||     10.2 |    10.0 |   -2%  ||    98.00 |    99.96 |   +2%  |  0.0000 |
+| 17      ||     28.6 |    26.2 |   -8%  ||    34.94 |    38.15 |   +9%  |  0.0000 |
+| 19      ||     10.0 |     8.3 |  -17%  ||    99.59 |   120.59 |  +21%  |  0.0000 |
+| 25      ||     15.4 |    14.5 |   -6%  ||    64.97 |    68.82 |   +6%  |  0.0000 |
 | 26      ||     16.0 |    16.2 |   +1%  ||    62.40 |    61.78 |   -1%  |  0.0000 |
+| 28      ||   1025.3 |   927.5 |  -10%  ||     0.98 |     1.08 |  +11%  |  0.0000 |
+| 29      ||     25.8 |    22.8 |  -11%  ||    38.82 |    43.80 |  +13%  |  0.0000 |
+| 31      ||    134.4 |   124.0 |   -8%  ||     7.44 |     8.06 |   +8%  |  0.0000 |
 | 34      ||     16.8 |    16.3 |   -3%  ||    59.47 |    61.45 |   +3%  |  0.0000 |
+| 35      ||     87.8 |    74.8 |  -15%  ||    11.39 |    13.36 |  +17%  |  0.0000 |
-| 39a     ||    178.1 |   191.0 |   +7%  ||     5.61 |     5.24 |   -7%  |  0.0000 |
-| 39b     ||    175.7 |   188.0 |   +7%  ||     5.69 |     5.32 |   -7%  |  0.0000 |
 | 41      ||     25.4 |    25.3 |   -0%  ||    39.40 |    39.55 |   +0%  |  0.0000 |
+| 42      ||      8.0 |     6.6 |  -18%  ||   125.59 |   152.58 |  +21%  |  0.0000 |
 | 43      ||    249.1 |   247.5 |   -1%  ||     4.01 |     4.04 |   +1%  |  0.0000 |
+| 45      ||      9.9 |     8.4 |  -15%  ||   101.41 |   119.27 |  +18%  |  0.0000 |
 | 48      ||     83.9 |    80.8 |   -4%  ||    11.92 |    12.37 |   +4%  |  0.0000 |
+| 50      ||     11.4 |     9.9 |  -13%  ||    87.94 |   100.83 |  +15%  |  0.0000 |
+| 52      ||      8.1 |     6.9 |  -14%  ||   123.66 |   144.36 |  +17%  |  0.0000 |
+| 55      ||      7.9 |     6.7 |  -15%  ||   126.92 |   148.69 |  +17%  |  0.0000 |
 | 62      ||     74.5 |    77.7 |   +4%  ||    13.42 |    12.88 |   -4%  |  0.0000 |
-| 65      ||     62.2 |    65.8 |   +6%  ||    16.09 |    15.21 |   -5%  |  0.0000 |
+| 69      ||     18.8 |    16.3 |  -13%  ||    53.16 |    61.33 |  +15%  |  0.0000 |
+| 73      ||     10.1 |     9.0 |  -11%  ||    98.87 |   110.86 |  +12%  |  0.0000 |
-| 79      ||     45.2 |    48.1 |   +6%  ||    22.13 |    20.80 |   -6%  |  0.0000 |
+| 81      ||     19.1 |    16.4 |  -14%  ||    52.26 |    60.82 |  +16%  |  0.0000 |
+| 83      ||      9.5 |     8.8 |   -7%  ||   105.42 |   113.61 |   +8%  |  0.0000 |
 | 85      ||     57.4 |    55.5 |   -3%  ||    17.42 |    18.02 |   +3%  |  0.0993 |
+| 88      ||     67.9 |    61.2 |  -10%  ||    14.72 |    16.33 |  +11%  |  0.0000 |
 | 91      ||     20.8 |    20.7 |   -0%  ||    48.03 |    48.22 |   +0%  |  0.0000 |
+| 93      ||    260.6 |   240.7 |   -8%  ||     3.84 |     4.16 |   +8%  |  0.0000 |
+| 96      ||      7.3 |     6.2 |  -15%  ||   136.35 |   160.78 |  +18%  |  0.0000 |
-| 97      ||    392.3 |   465.2 |  +19%  ||     2.55 |     2.15 |  -16%  |  0.0000 |
-| 99      ||    148.3 |   162.6 |  +10%  ||     6.74 |     6.15 |   -9%  |  0.0000 |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Sum     ||   3664.0 |  3583.7 |   -2%  ||          |          |        |         |
+| Geomean ||          |         |        ||          |          |   +7%  |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCDS - multi-threaded (50 clients)**
<details>
<summary>
Sum of avg. item runtimes: +3%
 ||
Geometric mean of throughput changes: -12%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+-----------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCDS_dcd55668b1db4e94834f448cc5c04498c50dda00_mt.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCDS_3004a139fc52f120d69984ae7b96b89a191e71f1_mt.json |
 +-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                          | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                          |
 |  benchmark_mode               | Ordered                                                                                                                                 | Ordered                                                                                                                                 |
 |  build_type                   | release                                                                                                                                 | release                                                                                                                                 |
 |  chunk_size                   | 65535                                                                                                                                   | 65535                                                                                                                                   |
 |  clients                      | 50                                                                                                                                      | 50                                                                                                                                      |
 |  compiler                     | gcc 9.2                                                                                                                                 | gcc 9.2                                                                                                                                 |
 |  cores                        | 0                                                                                                                                       | 0                                                                                                                                       |
 |  date                         | 2020-10-17 03:03:21                                                                                                                     | 2020-10-19 13:02:03                                                                                                                     |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                 | {'default': {'encoding': 'Dictionary'}}                                                                                                 |
 |  indexes                      | False                                                                                                                                   | False                                                                                                                                   |
 |  max_duration                 | 60000000000                                                                                                                             | 60000000000                                                                                                                             |
 |  max_runs                     | -1                                                                                                                                      | -1                                                                                                                                      |
 |  time_unit                    | ns                                                                                                                                      | ns                                                                                                                                      |
 |  using_scheduler              | True                                                                                                                                    | True                                                                                                                                    |
 |  utilized_cores_per_numa_node | [56, 0, 0, 0]                                                                                                                           | [56, 0, 0, 0]                                                                                                                           |
 |  verify                       | False                                                                                                                                   | False                                                                                                                                   |
 |  warmup_duration              | 0                                                                                                                                       | 0                                                                                                                                       |
 +-------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |     new |        ||      old |      new |        |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
-| 01      ||    197.4 |   254.2 |  +29%  ||   246.47 |   189.34 |  -23%  |  0.0000 |
-| 03      ||     25.3 |    47.0 |  +86%  ||  1618.52 |   925.51 |  -43%  |  0.0000 |
-| 06      ||    116.7 |   187.3 |  +60%  ||   410.79 |   256.60 |  -38%  |  0.0000 |
-| 07      ||    172.2 |   192.6 |  +12%  ||   281.68 |   250.98 |  -11%  |  0.0000 |
-| 09      ||    527.6 |   906.0 |  +72%  ||    91.45 |    53.80 |  -41%  |  0.0000 |
-| 10      ||     73.7 |    85.8 |  +16%  ||   643.18 |   549.84 |  -15%  |  0.0000 |
-| 13      ||    415.7 |   467.0 |  +12%  ||   118.09 |   104.78 |  -11%  |  0.0000 |
-| 15      ||     65.9 |   121.2 |  +84%  ||   706.68 |   386.00 |  -45%  |  0.0000 |
-| 17      ||    129.0 |   148.8 |  +15%  ||   373.06 |   324.05 |  -13%  |  0.0000 |
-| 19      ||     49.0 |    97.1 |  +98%  ||   935.71 |   483.47 |  -48%  |  0.0000 |
-| 25      ||     74.4 |   108.0 |  +45%  ||   635.70 |   440.28 |  -31%  |  0.0000 |
-| 26      ||     99.7 |   115.5 |  +16%  ||   478.79 |   409.78 |  -14%  |  0.0000 |
+| 28      ||   3127.6 |  2963.6 |   -5%  ||    15.18 |    15.97 |   +5%  |  0.3768 |
 | 29      ||    126.3 |   125.5 |   -1%  ||   380.06 |   383.04 |   +1%  |  0.4453 |
+| 31      ||    773.9 |   722.5 |   -7%  ||    63.44 |    67.73 |   +7%  |  0.0155 |
 | 34      ||     87.1 |    87.3 |   +0%  ||   542.72 |   541.88 |   -0%  |  0.7364 |
 | 35      ||    713.2 |   682.8 |   -4%  ||    68.96 |    72.01 |   +4%  |  0.0171 |
 | 39a     ||   1585.3 |  1603.9 |   +1%  ||    30.60 |    30.23 |   -1%  |  0.6670 |
 | 39b     ||   1576.1 |  1605.2 |   +2%  ||    30.53 |    30.28 |   -1%  |  0.5107 |
 | 41      ||   1384.0 |  1376.8 |   -1%  ||    34.03 |    33.43 |   -2%  |  0.9319 |
-| 42      ||     39.5 |    53.1 |  +34%  ||  1134.93 |   865.26 |  -24%  |  0.0000 |
 | 43      ||    764.6 |   755.6 |   -1%  ||    64.46 |    65.31 |   +1%  |  0.3699 |
+| 45      ||     49.5 |    43.0 |  -13%  ||   919.25 |  1048.62 |  +14%  |  0.0000 |
 | 48      ||    347.7 |   342.3 |   -2%  ||   140.94 |   143.24 |   +2%  |  0.3544 |
 | 50      ||    131.6 |   132.2 |   +0%  ||   365.31 |   363.61 |   -0%  |  0.6665 |
-| 52      ||     39.6 |    53.4 |  +35%  ||  1135.14 |   860.93 |  -24%  |  0.0000 |
-| 55      ||     37.0 |    52.1 |  +41%  ||  1204.52 |   879.06 |  -27%  |  0.0000 |
 | 62      ||    287.8 |   282.2 |   -2%  ||   170.14 |   173.22 |   +2%  |  0.0263 |
 | 65      ||    518.6 |   502.7 |   -3%  ||    95.05 |    97.87 |   +3%  |  0.0035 |
 | 69      ||     68.1 |    66.1 |   -3%  ||   687.46 |   707.62 |   +3%  |  0.0000 |
-| 73      ||     45.9 |    59.4 |  +29%  ||   996.30 |   780.92 |  -22%  |  0.0000 |
 | 79      ||    197.1 |   194.4 |   -1%  ||   246.79 |   250.36 |   +1%  |  0.0787 |
 | 81      ||    138.9 |   136.6 |   -2%  ||   347.08 |   352.60 |   +2%  |  0.0155 |
+| 83      ||     69.7 |    64.8 |   -7%  ||   671.58 |   719.82 |   +7%  |  0.0000 |
 | 85      ||    215.7 |   207.9 |   -4%  ||   226.27 |   234.71 |   +4%  |  0.0013 |
-| 88      ||    518.8 |   615.4 |  +19%  ||    93.11 |    79.23 |  -15%  |  0.0000 |
+| 91      ||     84.8 |    76.9 |   -9%  ||   559.20 |   613.95 |  +10%  |  0.0000 |
 | 93      ||   1302.7 |  1305.5 |   +0%  ||    37.54 |    37.41 |   -0%  |  0.9364 |
-| 96      ||     40.8 |    57.7 |  +41%  ||  1106.98 |   803.59 |  -27%  |  0.0000 |
 | 97      ||   3556.6 |  3521.8 |   -1%  ||    13.50 |    13.65 |   +1%  |  0.7420 |
 | 99      ||    743.7 |   734.8 |   -1%  ||    66.12 |    66.91 |   +1%  |  0.4375 |
 +---------++----------+---------+--------++----------+----------+--------+---------+
 | Sum     ||  20519.0 | 21156.0 |   +3%  ||          |          |        |         |
-| Geomean ||          |         |        ||          |          |  -12%  |         |
 +---------++----------+---------+--------++----------+----------+--------+---------+
```
</details>


**hyriseBenchmarkTPCC - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: +3%
 ||
Geometric mean of throughput changes: -4%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview-------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter        | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCC_dcd55668b1db4e94834f448cc5c04498c50dda00_st.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCC_3004a139fc52f120d69984ae7b96b89a191e71f1_st.json |
 +------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH        | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                         | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                         |
 |  benchmark_mode  | Shuffled                                                                                                                               | Shuffled                                                                                                                               |
 |  build_type      | release                                                                                                                                | release                                                                                                                                |
 |  chunk_size      | 65535                                                                                                                                  | 65535                                                                                                                                  |
 |  clients         | 1                                                                                                                                      | 1                                                                                                                                      |
 |  compiler        | gcc 9.2                                                                                                                                | gcc 9.2                                                                                                                                |
 |  cores           | 0                                                                                                                                      | 0                                                                                                                                      |
 |  date            | 2020-10-17 03:44:41                                                                                                                    | 2020-10-19 13:43:22                                                                                                                    |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}                                                                                                | {'default': {'encoding': 'Dictionary'}}                                                                                                |
 |  indexes         | False                                                                                                                                  | False                                                                                                                                  |
 |  max_duration    | 60000000000                                                                                                                            | 60000000000                                                                                                                            |
 |  max_runs        | -1                                                                                                                                     | -1                                                                                                                                     |
 |  scale_factor    | 1                                                                                                                                      | 1                                                                                                                                      |
 |  time_unit       | ns                                                                                                                                     | ns                                                                                                                                     |
 |  using_scheduler | False                                                                                                                                  | False                                                                                                                                  |
 |  verify          | False                                                                                                                                  | False                                                                                                                                  |
 |  warmup_duration | 0                                                                                                                                      | 0                                                                                                                                      |
 +------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Item         || Latency (ms/iter)  | Change || Throughput (iter/s) | Change |              p-value |
 |              ||      old |     new |        ||      old |      new |        |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
-| Delivery     ||     28.6 |    29.4 |   +3%  ||     3.77 |     3.58 |   -5%  | (run time too short) |
 | New-Order    ||     18.2 |    19.1 |   +5%  ||    42.21 |    40.44 |   -4%  | (run time too short) |
 | Order-Status ||      1.3 |     1.3 |   +0%  ||     3.77 |     3.60 |   -4%  | (run time too short) |
 | Payment      ||      2.5 |     2.6 |   +3%  ||    40.44 |    38.63 |   -4%  | (run time too short) |
 | Stock-Level  ||      4.1 |     4.1 |   -0%  ||     3.73 |     3.58 |   -4%  | (run time too short) |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Sum          ||     54.7 |    56.5 |   +3%  ||          |          |        |                      |
 | Geomean      ||          |         |        ||          |          |   -4%  |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
```
</details>


**hyriseBenchmarkTPCC - multi-threaded (50 clients)**
<details>
<summary>
Sum of avg. item runtimes: -1%
 ||
Geometric mean of throughput changes: -0%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview---------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter                     | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCC_dcd55668b1db4e94834f448cc5c04498c50dda00_mt.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkTPCC_3004a139fc52f120d69984ae7b96b89a191e71f1_mt.json |
 +-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH                     | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                         | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                         |
 |  benchmark_mode               | Shuffled                                                                                                                               | Shuffled                                                                                                                               |
 |  build_type                   | release                                                                                                                                | release                                                                                                                                |
 |  chunk_size                   | 65535                                                                                                                                  | 65535                                                                                                                                  |
 |  clients                      | 50                                                                                                                                     | 50                                                                                                                                     |
 |  compiler                     | gcc 9.2                                                                                                                                | gcc 9.2                                                                                                                                |
 |  cores                        | 0                                                                                                                                      | 0                                                                                                                                      |
 |  date                         | 2020-10-17 03:45:45                                                                                                                    | 2020-10-19 13:44:26                                                                                                                    |
 |  encoding                     | {'default': {'encoding': 'Dictionary'}}                                                                                                | {'default': {'encoding': 'Dictionary'}}                                                                                                |
 |  indexes                      | False                                                                                                                                  | False                                                                                                                                  |
 |  max_duration                 | 60000000000                                                                                                                            | 60000000000                                                                                                                            |
 |  max_runs                     | -1                                                                                                                                     | -1                                                                                                                                     |
 |  scale_factor                 | 1                                                                                                                                      | 1                                                                                                                                      |
 |  time_unit                    | ns                                                                                                                                     | ns                                                                                                                                     |
 |  using_scheduler              | True                                                                                                                                   | True                                                                                                                                   |
 |  utilized_cores_per_numa_node | [56, 0, 0, 0]                                                                                                                          | [56, 0, 0, 0]                                                                                                                          |
 |  verify                       | False                                                                                                                                  | False                                                                                                                                  |
 |  warmup_duration              | 0                                                                                                                                      | 0                                                                                                                                      |
 +-------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Item         || Latency (ms/iter)  | Change || Throughput (iter/s) | Change |              p-value |
 |              ||      old |     new |        ||      old |      new |        |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Delivery     ||    217.8 |   211.6 |   -3%  ||     4.44 |     4.54 |   +2%  | (run time too short) |
 |    unsucc.:  ||      3.6 |     3.9 |   +9%  ||   124.93 |   124.04 |   -1%  |                      |
 | New-Order    ||     98.2 |   100.3 |   +2%  ||    88.47 |    85.40 |   -3%  |               0.0220 |
 |    unsucc.:  ||      4.8 |     4.9 |   +2%  ||  1366.83 |  1361.29 |   -0%  |                      |
 | Order-Status ||      7.8 |     7.9 |   +1%  ||   129.37 |   128.60 |   -1%  |               0.0454 |
 | Payment      ||     17.0 |    18.7 |  +10%  ||     5.31 |     5.39 |   +2%  | (run time too short) |
 |    unsucc.:  ||      4.3 |     4.3 |   -1%  ||  1385.50 |  1377.10 |   -1%  |                      |
 | Stock-Level  ||     18.8 |    18.3 |   -2%  ||   129.36 |   128.57 |   -1%  |               0.0712 |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
 | Sum          ||    359.6 |   356.8 |   -1%  ||          |          |        |                      |
 | Geomean      ||          |         |        ||          |          |   -0%  |                      |
 +--------------++----------+---------+--------++----------+----------+--------+----------------------+
```
</details>


**hyriseBenchmarkJoinOrder - single-threaded**
<details>
<summary>
Sum of avg. item runtimes: +1%
 ||
Geometric mean of throughput changes: +2%
</summary>

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 | Parameter        | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkJoinOrder_dcd55668b1db4e94834f448cc5c04498c50dda00_st.json | /home/Markus.Dreseler/hyrise5/build-release/benchmark_all_results/hyriseBenchmarkJoinOrder_3004a139fc52f120d69984ae7b96b89a191e71f1_st.json |
 +------------------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
 |  GIT-HASH        | dcd55668b1db4e94834f448cc5c04498c50dda00-dirty                                                                                              | 3004a139fc52f120d69984ae7b96b89a191e71f1-dirty                                                                                              |
 |  benchmark_mode  | Ordered                                                                                                                                     | Ordered                                                                                                                                     |
 |  build_type      | release                                                                                                                                     | release                                                                                                                                     |
 |  chunk_size      | 65535                                                                                                                                       | 65535                                                                                                                                       |
 |  clients         | 1                                                                                                                                           | 1                                                                                                                                           |
 |  compiler        | gcc 9.2                                                                                                                                     | gcc 9.2                                                                                                                                     |
 |  cores           | 0                                                                                                                                           | 0                                                                                                                                           |
 |  date            | 2020-10-17 03:46:51                                                                                                                         | 2020-10-19 13:45:32                                                                                                                         |
 |  encoding        | {'default': {'encoding': 'Dictionary'}}                                                                                                     | {'default': {'encoding': 'Dictionary'}}                                                                                                     |
 |  indexes         | False                                                                                                                                       | False                                                                                                                                       |
 |  max_duration    | 60000000000                                                                                                                                 | 60000000000                                                                                                                                 |
 |  max_runs        | -1                                                                                                                                          | -1                                                                                                                                          |
 |  time_unit       | ns                                                                                                                                          | ns                                                                                                                                          |
 |  using_scheduler | False                                                                                                                                       | False                                                                                                                                       |
 |  verify          | False                                                                                                                                       | False                                                                                                                                       |
 |  warmup_duration | 0                                                                                                                                           | 0                                                                                                                                           |
 +------------------+---------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------+
```
</details>

```diff
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Item    || Latency (ms/iter)   | Change || Throughput (iter/s) | Change | p-value |
 |         ||      old |      new |        ||      old |      new |        |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | 10a     ||    357.7 |    346.3 |   -3%  ||     2.80 |     2.89 |   +3%  |  0.0000 |
 | 10b     ||    350.0 |    338.9 |   -3%  ||     2.86 |     2.95 |   +3%  |  0.0000 |
 | 10c     ||    938.3 |    934.8 |   -0%  ||     1.07 |     1.07 |   +0%  |  0.4328 |
+| 11a     ||    150.5 |    140.5 |   -7%  ||     6.64 |     7.12 |   +7%  |  0.0000 |
+| 11b     ||    141.1 |    131.1 |   -7%  ||     7.09 |     7.63 |   +8%  |  0.0000 |
 | 11c     ||    492.3 |    477.7 |   -3%  ||     2.03 |     2.09 |   +3%  |  0.0031 |
 | 11d     ||   4809.0 |   4904.4 |   +2%  ||     0.21 |     0.20 |   -2%  |  0.3032 |
 | 12a     ||    294.2 |    290.1 |   -1%  ||     3.40 |     3.45 |   +1%  |  0.1366 |
 | 12b     ||    634.7 |    622.3 |   -2%  ||     1.58 |     1.61 |   +2%  |  0.0407 |
 | 12c     ||   3635.2 |   3741.5 |   +3%  ||     0.28 |     0.27 |   -3%  |  0.0052 |
+| 13a     ||    179.5 |    170.1 |   -5%  ||     5.57 |     5.88 |   +5%  |  0.0000 |
+| 13b     ||    257.6 |    245.2 |   -5%  ||     3.88 |     4.08 |   +5%  |  0.0000 |
+| 13c     ||    137.2 |    127.2 |   -7%  ||     7.29 |     7.86 |   +8%  |  0.0000 |
+| 13d     ||    179.4 |    167.2 |   -7%  ||     5.57 |     5.98 |   +7%  |  0.0000 |
 | 14a     ||   4076.3 |   4155.9 |   +2%  ||     0.25 |     0.24 |   -2%  |  0.0479 |
 | 14b     ||   4699.3 |   4788.9 |   +2%  ||     0.21 |     0.21 |   -2%  |  0.0621 |
 | 14c     ||   4082.1 |   4155.7 |   +2%  ||     0.24 |     0.24 |   -2%  |  0.0540 |
 | 15a     ||    215.1 |    217.6 |   +1%  ||     4.65 |     4.60 |   -1%  |  0.0838 |
 | 15b     ||    212.1 |    209.7 |   -1%  ||     4.71 |     4.77 |   +1%  |  0.0145 |
-| 15c     ||    149.5 |    158.2 |   +6%  ||     6.69 |     6.32 |   -6%  |  0.0000 |
 | 15d     ||    458.0 |    457.6 |   -0%  ||     2.18 |     2.19 |   +0%  |  0.8280 |
-| 16a     ||   4797.4 |   5305.4 |  +11%  ||     0.21 |     0.19 |  -10%  |  0.0000 |
-| 16b     ||   6284.5 |   6790.1 |   +8%  ||     0.16 |     0.15 |   -7%  |       ˅ |
-| 16c     ||   4933.2 |   5372.0 |   +9%  ||     0.20 |     0.19 |   -8%  |  0.0000 |
-| 16d     ||   4906.2 |   5368.3 |   +9%  ||     0.20 |     0.19 |   -9%  |  0.0000 |
-| 17a     ||    946.5 |    993.6 |   +5%  ||     1.06 |     1.01 |   -5%  |  0.0000 |
 | 17b     ||    749.4 |    784.0 |   +5%  ||     1.33 |     1.28 |   -4%  |  0.0000 |
-| 17c     ||    706.0 |    742.2 |   +5%  ||     1.42 |     1.35 |   -5%  |  0.0000 |
 | 17d     ||    827.0 |    858.2 |   +4%  ||     1.21 |     1.17 |   -4%  |  0.0000 |
 | 17e     ||   2711.3 |   2765.9 |   +2%  ||     0.37 |     0.36 |   -2%  |  0.0063 |
-| 17f     ||   1518.2 |   1600.6 |   +5%  ||     0.66 |     0.62 |   -5%  |  0.0000 |
 | 18a     ||    596.6 |    588.9 |   -1%  ||     1.68 |     1.70 |   +1%  |  0.0148 |
+| 18b     ||    233.2 |    218.0 |   -6%  ||     4.29 |     4.59 |   +7%  |  0.0000 |
 | 18c     ||   3774.7 |   3826.3 |   +1%  ||     0.26 |     0.26 |   -1%  |  0.0250 |
 | 19a     ||   7845.8 |   7727.6 |   -2%  ||     0.13 |     0.13 |   +2%  |       ˅ |
+| 19b     ||   4805.1 |   4583.1 |   -5%  ||     0.21 |     0.22 |   +5%  |  0.0000 |
 | 19c     ||   7655.8 |   7512.9 |   -2%  ||     0.13 |     0.13 |   +2%  |       ˅ |
 | 19d     ||   5710.5 |   5798.1 |   +2%  ||     0.18 |     0.17 |   -2%  |  0.0769 |
 | 1a      ||     58.5 |     58.8 |   +1%  ||    17.10 |    17.01 |   -1%  |  0.0000 |
+| 1b      ||     22.2 |     20.4 |   -8%  ||    45.12 |    48.92 |   +8%  |  0.0000 |
+| 1c      ||     22.1 |     20.6 |   -7%  ||    45.17 |    48.43 |   +7%  |  0.0000 |
+| 1d      ||     21.9 |     20.4 |   -7%  ||    45.61 |    48.93 |   +7%  |  0.0000 |
 | 20a     ||   1235.0 |   1267.9 |   +3%  ||     0.81 |     0.79 |   -3%  |  0.0000 |
 | 20b     ||   1070.4 |   1099.0 |   +3%  ||     0.93 |     0.91 |   -3%  |  0.0000 |
 | 20c     ||   1100.4 |   1081.6 |   -2%  ||     0.91 |     0.92 |   +2%  |  0.0000 |
 | 21a     ||   3840.1 |   3908.0 |   +2%  ||     0.26 |     0.26 |   -2%  |  0.0000 |
 | 21b     ||    247.3 |    239.0 |   -3%  ||     4.04 |     4.18 |   +3%  |  0.0000 |
 | 21c     ||   3967.4 |   4045.9 |   +2%  ||     0.25 |     0.25 |   -2%  |  0.0000 |
 | 22a     ||   3481.7 |   3523.0 |   +1%  ||     0.29 |     0.28 |   -1%  |  0.1787 |
 | 22b     ||   3478.4 |   3526.5 |   +1%  ||     0.29 |     0.28 |   -1%  |  0.1262 |
 | 22c     ||   4097.9 |   4181.3 |   +2%  ||     0.24 |     0.24 |   -2%  |  0.0133 |
 | 22d     ||   4118.9 |   4215.4 |   +2%  ||     0.24 |     0.24 |   -2%  |  0.0044 |
-| 23a     ||    152.6 |    161.9 |   +6%  ||     6.55 |     6.17 |   -6%  |  0.0000 |
+| 23b     ||    121.7 |    111.2 |   -9%  ||     8.22 |     8.99 |   +9%  |  0.0000 |
 | 23c     ||    171.6 |    178.3 |   +4%  ||     5.83 |     5.61 |   -4%  |  0.0000 |
+| 24a     ||   4942.6 |   4681.8 |   -5%  ||     0.20 |     0.21 |   +6%  |  0.0000 |
 | 24b     ||   4835.7 |   4630.0 |   -4%  ||     0.21 |     0.22 |   +4%  |  0.0111 |
+| 25a     ||    236.2 |    217.6 |   -8%  ||     4.23 |     4.59 |   +9%  |  0.0000 |
+| 25b     ||    250.2 |    224.0 |  -10%  ||     4.00 |     4.46 |  +12%  |  0.0000 |
 | 25c     ||   3780.5 |   3851.9 |   +2%  ||     0.26 |     0.26 |   -2%  |  0.0003 |
 | 26a     ||    990.0 |    968.3 |   -2%  ||     1.01 |     1.03 |   +2%  |  0.0000 |
 | 26b     ||    978.4 |    964.5 |   -1%  ||     1.02 |     1.04 |   +1%  |  0.1301 |
 | 26c     ||    989.0 |    973.5 |   -2%  ||     1.01 |     1.03 |   +2%  |  0.0092 |
 | 27a     ||   3479.4 |   3546.0 |   +2%  ||     0.29 |     0.28 |   -2%  |  0.0019 |
 | 27b     ||   3454.5 |   3509.6 |   +2%  ||     0.29 |     0.28 |   -2%  |  0.0054 |
 | 27c     ||    200.5 |    196.8 |   -2%  ||     4.99 |     5.08 |   +2%  |  0.0171 |
 | 28a     ||   4135.2 |   4234.8 |   +2%  ||     0.24 |     0.24 |   -2%  |  0.3161 |
 | 28b     ||   3387.3 |   3442.0 |   +2%  ||     0.30 |     0.29 |   -2%  |  0.5785 |
 | 28c     ||   4135.6 |   4227.3 |   +2%  ||     0.24 |     0.24 |   -2%  |  0.3420 |
+| 29a     ||   4783.2 |   4572.7 |   -4%  ||     0.21 |     0.22 |   +5%  |  0.1755 |
+| 29b     ||    201.1 |    182.6 |   -9%  ||     4.97 |     5.48 |  +10%  |  0.0603 |
+| 29c     ||   4874.6 |   4661.3 |   -4%  ||     0.21 |     0.21 |   +5%  |  0.1927 |
+| 2a      ||     99.5 |     94.0 |   -6%  ||    10.05 |    10.64 |   +6%  |  0.0000 |
+| 2b      ||     81.9 |     76.9 |   -6%  ||    12.21 |    13.00 |   +6%  |  0.0000 |
+| 2c      ||     53.9 |     50.4 |   -6%  ||    18.56 |    19.83 |   +7%  |  0.0000 |
 | 2d      ||    124.4 |    121.6 |   -2%  ||     8.04 |     8.23 |   +2%  |  0.0000 |
+| 30a     ||    251.9 |    232.2 |   -8%  ||     3.97 |     4.31 |   +8%  |  0.0000 |
 | 30b     ||   1098.9 |   1083.9 |   -1%  ||     0.91 |     0.92 |   +1%  |  0.4896 |
 | 30c     ||   3805.1 |   3921.3 |   +3%  ||     0.26 |     0.26 |   -3%  |  0.1812 |
+| 31a     ||    287.8 |    258.4 |  -10%  ||     3.47 |     3.87 |  +11%  |  0.0000 |
 | 31b     ||   1128.4 |   1095.5 |   -3%  ||     0.89 |     0.91 |   +3%  |  0.0113 |
+| 31c     ||    290.9 |    257.2 |  -12%  ||     3.44 |     3.89 |  +13%  |  0.0000 |
+| 32a     ||     36.8 |     34.3 |   -7%  ||    27.20 |    29.14 |   +7%  |  0.0000 |
 | 32b     ||    267.8 |    273.0 |   +2%  ||     3.73 |     3.66 |   -2%  |  0.0000 |
+| 33a     ||     70.2 |     62.9 |  -10%  ||    14.24 |    15.89 |  +12%  |  0.0000 |
+| 33b     ||     69.3 |     62.3 |  -10%  ||    14.42 |    16.06 |  +11%  |  0.0000 |
+| 33c     ||     84.7 |     75.8 |  -11%  ||    11.81 |    13.19 |  +12%  |  0.0000 |
 | 3a      ||   3910.0 |   3997.9 |   +2%  ||     0.26 |     0.25 |   -2%  |  0.0000 |
+| 3b      ||     35.6 |     31.4 |  -12%  ||    28.07 |    31.88 |  +14%  |  0.0000 |
 | 3c      ||   4777.5 |   4872.5 |   +2%  ||     0.21 |     0.21 |   -2%  |  0.0000 |
+| 4a      ||    114.5 |    109.3 |   -5%  ||     8.73 |     9.15 |   +5%  |  0.0000 |
+| 4b      ||     26.7 |     23.5 |  -12%  ||    37.38 |    42.49 |  +14%  |  0.0000 |
-| 4c      ||    126.4 |    134.1 |   +6%  ||     7.91 |     7.46 |   -6%  |  0.0000 |
 | 5a      ||   3781.6 |   3872.2 |   +2%  ||     0.26 |     0.26 |   -2%  |  0.0000 |
 | 5b      ||    276.5 |    283.2 |   +2%  ||     3.62 |     3.53 |   -2%  |  0.0000 |
 | 5c      ||   4384.2 |   4455.4 |   +2%  ||     0.23 |     0.22 |   -2%  |  0.0000 |
+| 6a      ||    228.0 |    212.1 |   -7%  ||     4.39 |     4.71 |   +8%  |  0.0000 |
 | 6b      ||    958.1 |    971.8 |   +1%  ||     1.04 |     1.03 |   -1%  |  0.0000 |
+| 6c      ||    227.7 |    212.3 |   -7%  ||     4.39 |     4.71 |   +7%  |  0.0000 |
 | 6d      ||    958.2 |    971.2 |   +1%  ||     1.04 |     1.03 |   -1%  |  0.0000 |
+| 6e      ||    227.3 |    212.4 |   -7%  ||     4.40 |     4.71 |   +7%  |  0.0000 |
 | 6f      ||   1420.0 |   1430.3 |   +1%  ||     0.70 |     0.70 |   -1%  |  0.0000 |
+| 7a      ||    278.1 |    236.3 |  -15%  ||     3.60 |     4.23 |  +18%  |  0.0000 |
+| 7b      ||    136.9 |    103.4 |  -24%  ||     7.31 |     9.67 |  +32%  |  0.0000 |
 | 7c      ||  10247.3 |  10615.3 |   +4%  ||     0.10 |     0.09 |   -3%  |       ˅ |
+| 8a      ||    111.9 |     97.3 |  -13%  ||     8.94 |    10.27 |  +15%  |  0.0000 |
 | 8b      ||    157.3 |    151.1 |   -4%  ||     6.36 |     6.62 |   +4%  |  0.0000 |
 | 8c      ||   3951.7 |   4093.6 |   +4%  ||     0.25 |     0.24 |   -3%  |  0.0000 |
 | 8d      ||   1263.3 |   1311.9 |   +4%  ||     0.79 |     0.76 |   -4%  |  0.0000 |
 | 9a      ||   3451.8 |   3443.5 |   -0%  ||     0.29 |     0.29 |   +0%  |  0.6645 |
+| 9b      ||    313.5 |    299.6 |   -4%  ||     3.19 |     3.34 |   +5%  |  0.0000 |
 | 9c      ||   3437.7 |   3464.2 |   +1%  ||     0.29 |     0.29 |   -1%  |  0.1367 |
 | 9d      ||   4044.4 |   4038.8 |   -0%  ||     0.25 |     0.25 |   +0%  |  0.8067 |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 | Sum     || 214910.1 | 217705.1 |   +1%  ||          |          |        |         |
 | Geomean ||          |          |        ||          |          |   +2%  |         |
 +---------++----------+----------+--------++----------+----------+--------+---------+
 |         || ˅ Insufficient number of runs for p-value calculation                  |
 +---------++----------+----------+--------++----------+----------+--------+---------+
```
</details>